src/main/java/org/broadinstitute/hellbender/tools/exome/ExonCollections.java

```diff
@@ -0,0 +1,108 @@
+        return new HashedListExonCollection<SimpleInterval>(intervals) {
+
+            @Override
+            public String name(final Object exon) {
+                Utils.nonNull(exon,"the input exon cannot be null");
+                if (exon instanceof SimpleInterval) {
+                    return exon.toString();
+                } else {
+                    return null;
+                }
+            }
+
+            @Override
+            public SimpleInterval location(final Object exon) {
+                Utils.nonNull(exon,"the input exon cannot be null");
+                if (exon instanceof SimpleInterval) {
+                    return (SimpleInterval) exon;
+                } else {
+                    return null;
+                }
+            }
+        };
+    }
+
+     /**
+     * Creates an exon collection out from a BED feature list.
+     * <p>
+     *     Each feature in the input bed feature list will correspond to an exon whose name is the name of the BED
+     *     feature as returned by {@link BEDFeature#getName()}.
+     * </p>
+     *
+     * @param features the exon features.
+     * @param <T> the specific {@link BEDFeature} subclass.
+     * @throws IllegalArgumentException if {@code features} is {@code null}, it contains the {@code null} element or
+     *     overlapping features.
+     *
+     * @return never {@code null}.
+     */
+    public static <T extends BEDFeature> ExonCollection<T> fromBEDFeatureList(final List<T> features) {
+        return new HashedListExonCollection<T>(Utils.nonNull(features,"the input feature list cannot be null")) {
+            @Override
+            public String name(final Object exon) {
+                if (exon == null) {
+                    throw new IllegalArgumentException("the input exon cannot be null");
+                } else if (exon instanceof BEDFeature) {
+                    final String name = ((BEDFeature)exon).getName();
+                    return name != null && !name.isEmpty() ? name : null;
+                } else {
+                    return null;
+                }
+            }
+        };
+    }
+
+    /**
+     * Creates an exon collection out from a BED feature containing file.
+     *
+     * @param file the exon feature file.
+     * @param <T> the specific {@link BEDFeature} subclass.
+     * @throws IllegalArgumentException if {@code features} is {@code null}, it contains the {@code null} element or
+     *     overlapping features.
+     *
+     * @return never {@code null}.
+     */
+    public static <T extends BEDFeature> ExonCollection<T> fromBEDFeatureFile(final File file, final FeatureCodec<T, ?> codec) {
+        Utils.nonNull(file,"the input file cannot be null");
+        Utils.nonNull(codec,"the codec cannot be null");
+        final FeatureDataSource<T> source = new FeatureDataSource<>(file,codec);
+        final List<T> features = StreamSupport.stream(source.spliterator(),false).collect(Collectors.toList());
```

Comment:
Not in practice. < 200K exon for Human. It takes far longer to deal with the BAM index preprocessing... 

If would be more inefficient if we start scattering to small subsections of the exome... and in that case we may well want to revisit this but it works for now. 
We may want to revisit the design of this walker in the future  but I rather get